### PR TITLE
[*] CORE : You can now search with filter by category

### PR DESCRIPTION
**Pull request by wellic**
Added ability to filter products by category into the method Search::Find. This allows to make the search more flexible using the id_category the same as in Products::getProducts.

### CODE REVIEW

**Comment by wellic** [comment]
I have tried several ways to send a pull request. Each time the error. How to properly do commit to pull request?

**Comment by kpodemski** [comment]
I think this is ok for develop branch :)

**Comment by wellic** [comment]
I tried on '1.6' - branch and 'develop'-branch. The test was failed.

**Comment by kpodemski** [comment]
These tests are not related to your change from what I see

**Comment by wellic** [comment]
I used this guide: https://github.com/PrestaShop/PrestaShop/blob/develop/CONTRIBUTING.md

What I do wrong?

**Comment by kpodemski** [comment]
Everything is fine with your commit, tests have failed not because of your code.

**Comment by wellic** [comment]
Should I do something this pull request?

**Comment by kpodemski** [comment]
No, it's ok, later someone in charge will check this PR and decide whether this can be merged or not.

File: classes/Search.php [hunk 1]
```diff
@@ -170,6 +170,11 @@ public static function sanitize($string, $id_lang, $indexation = false, $iso_cod
     }
 
     public static function find($id_lang, $expr, $page_number = 1, $page_size = 1, $order_by = 'position',
+        $order_way = 'desc', $ajax = false, $use_cookie = true, Context $context = null) {
+        return self::findWithCategory($id_lang, $expr, false, $page_number, $page_size, $order_by, $order_way, $ajax, $use_cookie);
```
Comment (PrestaEdit) [inline classes/Search.php]: Missing the $context var in params, at the end, I think.

**Comment by wellic** [comment]
Fixed in #4376.

**Pull request closed without merging**